index/scorch/optimize_knn.go

```diff
@@ -114,13 +114,14 @@ func (o *OptimizeVR) Finish() error {
 								eligibleVectorInternalIDs.And(snapshotGlobalDocNums[index])
 							}
 
-							eligibleLocalDocNums := make([]uint64,
-								eligibleVectorInternalIDs.GetCardinality())
+							eligibleLocalDocNums := roaring.NewBitmap()
 							// get the (segment-)local document numbers
-							for i, docNum := range eligibleVectorInternalIDs.ToArray() {
+							for _, docNum := range eligibleVectorInternalIDs.ToArray() {
+								// TODO: IS THIS CONVERSION NECESSARY?
+								// can we not simply use eligibleVectorInternalIDs or a copy here?
```

Comment:
About this conversion, it's converting a global(index-wide) doc num to a local(to the segment) doc number, so I don't think it can be removed.

Reply:
Ack